### PR TITLE
feat(server): add watermarking tests

### PR DESCRIPTION
**Pull request by ehsanmok**
Thank you for providing such a high quality library. This PR contains some improvements that I've found while skimming through the code.

### CODE REVIEW

**Review by OlivierDehaene — COMMENTED** [review]
Thanks!

File: router/src/health.rs [hunk 1]
```diff
@@ -4,6 +4,9 @@ use text_generation_client::{
     Batch, NextTokenChooserParameters, Request, ShardedClient, StoppingCriteriaParameters,
 };
 
+const LIVENESS_ID: u64 = u64::MAX;
+const BATCH_ID: u64 = u64::MAX - 1;
```
Comment (OlivierDehaene) [inline router/src/health.rs:8]: why -1?

File: server/pyproject.toml [hunk 2]
```diff
@@ -35,6 +35,9 @@ bnb = ["bitsandbytes"]
 grpcio-tools = "^1.51.1"
 pytest = "^7.3.0"
 
+[tool.pytest.ini_options]
```
Comment (OlivierDehaene) [inline server/pyproject.toml:38]: We don't have asyncio tests in the server.

File: router/src/health.rs (hunk 1 above)
Comment (ehsanmok) [reply router/src/health.rs:8]: Avoiding collision of special ids?

File: server/pyproject.toml (hunk 2 above)
Comment (ehsanmok) [reply server/pyproject.toml]: Ah! good catch

**Pull request merged**